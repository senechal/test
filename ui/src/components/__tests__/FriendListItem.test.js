import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import FriendListItem from '../FriendListItem';


describe('Tests for FriendListItem', () => {
  let props;
  beforeEach(() => {
    props = {
      id: '0',
      name: 'mock',
      starred: true,
      sex: 'male',
      starFriend: sinon.spy(),
      deleteFriend:sinon.spy(),
    }
  });
  configure({ adapter: new Adapter() });
  it('should render without crashing', () => {
    const wrapper = shallow(<FriendListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing starred', () => {
    props.starred = true;
    const wrapper = shallow(<FriendListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing female', () => {
    props.sex = 'female';
    const wrapper = shallow(<FriendListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call starFriend on button click', () => {
    const wrapper = shallow(<FriendListItem {...props}/>);
    const buttons = wrapper.find('button');
    buttons.at(0).simulate('click')
    expect(props.starFriend.called).toBeTruthy();
    expect(props.starFriend.calledWith('0', false)).toBeTruthy();
  });
  it('should call deleteFriend on button click', () => {
    const wrapper = shallow(<FriendListItem {...props}/>);
    const buttons = wrapper.find('button');
    buttons.at(1).simulate('click')
    expect(props.deleteFriend.called).toBeTruthy();
    expect(props.deleteFriend.calledWith('0')).toBeTruthy();
  });
});