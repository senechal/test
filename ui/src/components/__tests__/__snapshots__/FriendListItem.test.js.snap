// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FriendListItem should render without crashing 1`] = `
<li>
  <div>
    <div>
      <span>
        mock
      </span>
      <i
        className="fa fa-mars"
        title="Male"
      />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-star"
      />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </div>
</li>
`;

exports[`Tests for FriendListItem should render without crashing female 1`] = `
<li>
  <div>
    <div>
      <span>
        mock
      </span>
      <i
        className="fa fa-venus"
        title="female"
      />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-star"
      />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </div>
</li>
`;

exports[`Tests for FriendListItem should render without crashing starred 1`] = `
<li>
  <div>
    <div>
      <span>
        mock
      </span>
      <i
        className="fa fa-mars"
        title="Male"
      />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-star"
      />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </div>
</li>
`;
